### PR TITLE
Fix Button issues

### PR DESCRIPTION
**Pull request by ssrva**
- Move padding from ButtonContent to ButtonBase
- Add text-overflow for ButtonContent

### CODE REVIEW

File: src/components/Button/components/ButtonBase.tsx [hunk 1]
```diff
@@ -104,11 +117,51 @@ const getBoxShadow = (props: IStyledBaseButton) => {
   return `0px 0px 2px ${knitui.shades.blue50}`
 }
 
+const getHorizontalPadding = (props: IStyledBaseButton) => {
+  const {
+    customProps: { size },
+  } = props
+  const paddings = {
+    small: 0.7,
+    medium: 1,
```
Comment (anshumanv) [inline src/components/Button/components/ButtonBase.tsx:126]: sure both med and large should have same padding?

**Review by anshumanv — APPROVED** [review]

**Comment by anshumanv** [comment]
Also inset if f'd 
image

**Comment by anshumanv** [comment]
Lets not apply auto ellipsis instead let the button take up the space needed by text

image

File: src/components/Button/components/ButtonBase.tsx (hunk 1 above)
Comment (ssrva) [reply src/components/Button/components/ButtonBase.tsx:126]: It was actually copied over from ButtonContent, let me see if increased padding is needed

Comment (anshumanv) [reply src/components/Button/components/ButtonBase.tsx:126]: nah then it should be as per the design, you can leave it be

**Comment by ssrva** [comment]
> Lets not apply auto ellipsis instead let the button take up the space needed by text
> 
> image

It was an attempt to make things responsive. It seemed to break other things. Reverted it

**Pull request merged**